### DISCUSSION TITLE
click play will change wave scale

### DISCUSSION THREAD

**Discussion by yg2342423523** [katspaugh/wavesurfer.js#2871] [Q&A]
hello my hero,i set my wave use example json [https://wavesurfer-js.org/examples/#predecoded.js](url) to load,but i found that only when i  click the wave, wave will be correct size,check this pic
![06706a551264ff6b8db6db3b799608d](https://github.com/katspaugh/wavesurfer.js/assets/102865541/e82f710e-a3db-4fad-b534-acdc948aeb11)
![5fdbf916b6ad61472f2d13a4ba01bf6](https://github.com/katspaugh/wavesurfer.js/assets/102865541/fd6e73af-c24f-4df8-812b-b03509ffd00d)

this is my code:    

`  let waveEl = document.createElement('div')
      waveEl.classList.add('waveform')
      // let test = document.getElementById("test");
      var wavesurfer = WaveSurfer.create({
        container: waveEl,
        waveColor: '#ccc',
        progressColor: '#0091ff',
        height: 130,
        barMinHeight: 1,
        cursorWidth: 0,
      });
      wavesurfer.load(window.URL.createObjectURL(url), readPeaks(audioName));`

when i set the height will change its scale:
![e58d59bdda88cc6fbdcc796e2c9f14a](https://github.com/katspaugh/wavesurfer.js/assets/102865541/8ddcd709-d1fc-443b-818f-934dd329b62b)

when i give it barwidth,it will disapper
 i have this problem for one week,i have no idea with that,hope you can help me<thank you

**Comment by katspaugh** [comment]
When are you adding the waveEl to the DOM? Try doing it before you create a wavesurfer instance.

**Comment by yg2342423523** [comment]
thank you，sir！It works！It must be a crazy problem. Make a dom before create waves!
